### PR TITLE
Accounts and Database Integration

### PR DESCRIPTION
**Pull request by juandpr**
I created 2 new forms, one to create a new account and one with the information of the current logged in account.

### CODE REVIEW

File: CODIGO/Declares.bas [hunk 1]
```diff
@@ -717,10 +721,32 @@ Global OtroInventario(1 To MAX_INVENTORY_SLOTS) As Inventory
 
 Public UserHechizos(1 To MAXHECHI) As Integer
 
+Public Type PjCuenta
+    nombre      As String
+    Head        As Integer
+    Body        As Integer
+    shield      As Byte
```
Comment (RecoX) [inline CODIGO/Declares.bas]: The variables should start with uppercase to maintain a normalization around the code?

File: CODIGO/General.bas [hunk 2]
```diff
@@ -234,43 +234,38 @@ Function AsciiValidos(ByVal cad As String) As Boolean
...
+        CharAscii = Asc(mid$(AccountPassword, LoopC, 1))
         If Not LegalCharacter(CharAscii) Then
             MsgBox ("Password invalido. El caracter " & Chr$(CharAscii) & " no esta� permitido.")
             Exit Function
         End If
     Next LoopC
     
-    If UserName = "" Then
-        MsgBox ("Ingrese un nombre de personaje.")
+    If AccountName = "" Then
```
Comment (RecoX) [inline CODIGO/General.bas]: Add TODO comment for create a REGEX for email validations.

File: CODIGO/frmConnect.frm [hunk 3]
```diff
@@ -275,6 +283,10 @@ Private cBotonTeclas As clsGraphicalButton
 
 Public LastButtonPressed As clsGraphicalButton
 
+Private Sub cmdNuevaCuenta_Click()
+frmCrearCuenta.Show
```
Comment (RecoX) [inline CODIGO/frmConnect.frm]: Indentation

File: CODIGO/frmConnect.frm [hunk 4]
```diff
@@ -513,27 +523,7 @@ Private Sub imgConectarse_Click()
...
-    frmMain.Socket1.Connect
-#Else
-    If frmMain.Winsock1.State <> sckClosed Then
-        frmMain.Winsock1.Close
-        DoEvents
-    End If
-    frmMain.Winsock1.Connect CurServerIp, CurServerPort
-#End If
-
+MsgBox ("Usar cuenta")
```
Comment (RecoX) [inline CODIGO/frmConnect.frm]: Indentation

File: CODIGO/frmCrearPersonaje.frm [hunk 5]
```diff
@@ -1477,15 +1481,15 @@ Private Sub CargarCombos()
 End Sub
 
 Function CheckData() As Boolean
-    If txtPasswd.Text <> txtConfirmPasswd.Text Then
-        MsgBox "Los passwords que tipeo no coinciden, por favor vuelva a ingresarlos."
-        Exit Function
-    End If
+    'If txtPasswd.Text <> txtConfirmPasswd.Text Then
```
Comment (RecoX) [inline CODIGO/frmCrearPersonaje.frm]: Delete unused code instead of comment it?

File: CODIGO/frmCrearPersonaje.frm [hunk 6]
```diff
@@ -1600,17 +1610,15 @@ Private Sub imgCrear_Click()
...
-    For i = 1 To Len(UserPassword)
-        CharAscii = Asc(mid$(UserPassword, i, 1))
-        If Not LegalCharacter(CharAscii) Then
-            MsgBox ("Password inv�lido. El caract�r " & Chr$(CharAscii) & " no est� permitido.")
-            Exit Sub
-        End If
-    Next i
-    
-    UserEmail = txtMail.Text
+    'UserPassword = txtPasswd.Text
```
Comment (RecoX) [inline CODIGO/frmCrearPersonaje.frm]: Delete unused code instead of comment it ?

File: CODIGO/frmCuenta.frm [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    End If
+End If
+End Sub
+
+Private Sub cmdCrear_Click()
+    frmCrearPersonaje.Show
+End Sub
+
+Private Sub Form_Load()
+Dim i As Byte
```
Comment (RecoX) [inline CODIGO/frmCuenta.frm]: Indentation of this function

File: CODIGO/frmPanelAccount.frm [hunk 8]
```diff
@@ -0,0 +1,652 @@
...
+For i = 1 To 10
+  If lblAccData(i).Caption = "" Then
+     frmCrearPersonaje.Show
+     Exit Sub
+  End If
+Next i
+End Sub
+
+Private Sub imgSalir_Click()
+frmMain.Socket1.Disconnect
```
Comment (RecoX) [inline CODIGO/frmPanelAccount.frm]: Indentation

File: CODIGO/mDx8_Engine.bas [hunk 9]
```diff
@@ -729,7 +729,80 @@ Public Sub Engine_Update_FPS()
...
+    init_y = 20
+   
+    Dim Light(3) As Long
+    Light(0) = D3DColorXRGB(255, 255, 255)
+    Light(1) = D3DColorXRGB(255, 255, 255)
+    Light(2) = D3DColorXRGB(255, 255, 255)
+    Light(3) = D3DColorXRGB(255, 255, 255)
+    
+    If cPJ(Index).Race = eRaza.Humano Or cPJ(Index).Race = eRaza.Elfo Or cPJ(Index).Race = eRaza.ElfoOscuro Then
+        head_offset = 8
```
Comment (RecoX) [inline CODIGO/mDx8_Engine.bas]: what is the 8? can we create a constant to understand the meaning of the 8 ?

Comment (juandpr) [reply CODIGO/mDx8_Engine.bas]: That's the head offset, yes we can create a constant

File: CODIGO/Declares.bas (hunk 1 above)
Comment (juandpr) [reply CODIGO/Declares.bas]: If we do so, we should change the name in other places. Because VB will replace them

**Comment by juandpr** [comment]
@RecoX I think so to, but I am not gonna do it. At least in this PR

**Pull request merged**